### PR TITLE
More `HUB` debugging

### PR DESCRIPTION
**Pull request by OlivierBBB**

### CODE REVIEW

File: hub/constraints/generalities/program_counter.lisp [hunk 1]
```diff
@@ -11,6 +11,11 @@
...
+(defconstraint   generalities---program-counter---PC_NEW-vanishes-upon-stack-exception (:guard PEEK_AT_STACK)
+                 (if-not-zero    (force-bool   (+    stack/SUX    stack/SOX))
+                                 (vanishes!    PC_NEW)))
+
 (defconstraint   generalities---program-counter---automatic-update (:guard PEEK_AT_STACK)
-                 (if-zero (force-bin (+ stack/PUSHPOP_FLAG stack/JUMP_FLAG))
-                          (eq! PC_NEW (+ 1 PC))))
+                 (if-zero    (force-bool   (+    stack/SUX    stack/SOX))
+                             (if-zero    (force-bin (+ stack/PUSHPOP_FLAG stack/JUMP_FLAG))
+                                         (eq! PC_NEW (+ 1 PC)))))
```
Comment (OlivierBBB) [inline hub/constraints/generalities/program_counter.lisp:21]: Update of PC_NEW constraints

File: hub/constraints/instruction-handling/halting/return.lisp [hunk 2]
```diff
@@ -266,38 +266,38 @@
...
+                                                                                ;; src_offset_hi                                  ;; source offset high
+                                                                                (return-instruction---offset-lo)                  ;; source offset low
+                                                                                ;; tgt_offset_lo                                  ;; target offset low
+                                                                                ;; size                                           ;; size
+                                                                                ;; ref_offset                                     ;; reference offset
+                                                                                ;; ref_size                                       ;; reference size
+                                                                                (- 1 (return-instruction---exception-flag-ICPX))  ;; success bit; this double negation stuff will be resolved by spec issue #715
```
Comment (OlivierBBB) [inline hub/constraints/instruction-handling/halting/return.lisp:279]: This was a bug

Comment (lorenzogentile404) [reply hub/constraints/instruction-handling/halting/return.lisp:279]: shouldn't the definition of `set-MMU-instruction---invalid-code-prefix` change too?

Comment (OlivierBBB) [reply hub/constraints/instruction-handling/halting/return.lisp:279]: No. The issue was that we were not respecting the HUB/MMU interface. We were feeding the "failure bit" in the HUB and expecting the "success bit" in the MMU.

**Pull request merged**